src/app/child-dev-project/attendance/add-day-attendance/add-day-attendance.component.spec.ts

```diff
@@ -37,4 +47,58 @@ describe("AddDayAttendanceComponent", () => {
   it("should create", () => {
     expect(component).toBeTruthy();
   });
+
+  it("should sort event participants by schoolClass of students", async () => {
+    const testStudents = [
+      new Child("0"),
+      new Child("1"),
+      new Child("2"),
+      new Child("3"),
+      new Child("4"),
+      new Child("5"),
+    ];
+    testStudents[0].schoolClass = "10";
+    testStudents[1].schoolClass = "9";
+    testStudents[2].schoolClass = "KG";
+    testStudents[3].schoolClass = undefined;
+    testStudents[4].schoolClass = "9";
+    testStudents[5].schoolClass = "1";
+
+    component.attendanceType = "coaching";
+    component.day = new Date();
+
+    component.finishStudentSelectionStage(testStudents);
+
+    const actualStudentClassOrder = component.event.children.map((childId) => {
+      const child = testStudents.find((c) => c.getId() === childId);
+      return child.schoolClass;
+    });
+    const expectedStudentClassOrder = ["1", "9", "9", "10", "KG", undefined];
+    expect(JSON.stringify(actualStudentClassOrder)).toEqual(
```

Comment:
it is not necessary, but if the order is wrong the fail message is very complicated to interpret (expect "1" received "9" at index 0 ...) because it doesn't show you the overall order